sdk/servicebus/azservicebus/internal/sender.go

```diff
@@ -49,21 +49,27 @@ type (
 		cancelAuthRefresh func() <-chan struct{}
 	}
 
-	// SendOption provides a way to customize a message on sending
-	SendOption func(event *Message) error
-
 	eventer interface {
 		toMsg() (*amqp.Message, error)
 		GetKeyValues() map[string]interface{}
 		Set(key string, value interface{})
 	}
-
-	// SenderOption provides a way to customize a Sender
-	SenderOption func(*Sender) error
 )
 
+// ie: `*internal.Sender`
+type LegacySender interface {
+	SendAMQPMessage(ctx context.Context, msg *amqp.Message) error
+	Close(ctx context.Context) error
+	MaxMessageSize() uint64
+}
+
+// NewLegacySender is just a wrapper for `NewSender` that returns an interface. Makes testing simpler.
```

Comment:
Is this also something we expect customers to use? Or is this strictly for testing?

Reply:
Definitely just for me. It's in the `internal` package, so not usable outside of our module.